core/src/calleeFunctions/helpers/oneInch.ts

```diff
@@ -0,0 +1,161 @@
+import { ethers } from 'ethers';
+import { getCalleeAddressByCollateralType } from '../../constants/CALLEES';
+import { getCollateralConfigBySymbol } from '../../constants/COLLATERALS';
+import { getErc20SymbolByAddress } from '../../contracts';
+import { getChainIdByNetworkType, getNetworkConfigByType } from '../../network';
+import { CollateralConfig } from '../../types';
+import BigNumber from '../../bignumber';
+import { getTokenAddressByNetworkAndSymbol } from '../../tokens';
+import { WAD_NUMBER_OF_DIGITS } from '../../constants/UNITS';
+import { RateLimiter } from 'limiter';
+
+const REQUEST_LIMITER = new RateLimiter({ tokensPerInterval: 1, interval: 'second' });
```

Comment:
> This will lead to the situation when as soon as we have more than 1 auction, only the first in the list get enriched with 1inch data, all others throw the error.

could you reference the source of this information?

because from the [source](https://github.com/jhurliman/node-rate-limiter/blob/9bbd3b036c33e974d51d7b40ddeee13e71bb69b0/src/RateLimiter.ts#L51) and description on npm i can't find any support of this claim. It only throws if we requrest more than max amount of tokens it seems. I.e. if we request 2 or more tokens to be deducted.

Reply:
replaced with the suggested package.

Reply:
> could you reference the source of this information?

First example in the readme states that ["This call will throw if we request more than the maximum number of requests"](https://github.com/jhurliman/node-rate-limiter/blob/9bbd3b036c33e974d51d7b40ddeee13e71bb69b0/README.md?plain=1#L30)

Reply:
```js

const limiter = new RateLimiter(maxAmount);

for (const i of Array(maxAmount + 1).keys()) {
  await  limiter.removeTokens(1)
} // this works

await limiter.removeTokens(maxAmount + 1) // this throws
```

Reply:
Not sure why do you bring this code snippet ,since it doesn't contradict the overall point: the package you've initially used throws, if limit is reached instead of waiting for the previous calls to finish. Thanks for switching to `async-await-queue` 🙂